compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -576,10 +577,20 @@ impl<'a> Parser<'a> {
     /// Note that this does *not* parse bare trait objects.
     fn parse_dyn_ty(&mut self, impl_dyn_multi: &mut bool) -> PResult<'a, TyKind> {
         self.bump(); // `dyn`
+
+        // parse dyn* types
+        let dyn_star = matches!(self.token.kind, TokenKind::BinOp(token::Star));
+        let syntax = if dyn_star {
+            self.bump(); // `*`
```

Comment:
nit: I think you can replace this with a call to `eat` instead of doing a `matches!` then a `bump`.

Reply:
Done